Riot/Modules/Home/AllChats/AllChatsLayoutSettingsManager.swift

```diff
@@ -0,0 +1,83 @@
+// 
+// Copyright 2022 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+
+// MARK: - Notification constants
+
+extension AllChatsLayoutSettingsManager {
+    /// Posted if settings have changed
+    public static let willUpdateSettings = Notification.Name("AllChatLayoutSettingsManagerWillUpdateSettings")
+    public static let didUpdateSettings = Notification.Name("AllChatLayoutSettingsManagerDidUpdateSettings")
+}
+
+@objcMembers
+final class AllChatsLayoutSettingsManager: NSObject {
+    
+    static let shared = AllChatsLayoutSettingsManager()
+    private var notifyChanges: Bool = true
```

Comment:
`notifyChanges` is needed to avoid extra notifications when `AllChatsLayoutSettings.activeFilters` is updated as the instance needs to be saved by the `AllChatsLayoutSettingsManager` without notifying the settings have changed.

It was not clear indeed and I've solved it by moving `activeFilters` from `AllChatsLayoutSettings` to `AllChatsLayoutSettingsManager`.